core/store/models/common.go

```diff
@@ -183,32 +183,53 @@ func (j JSON) Bytes() []byte {
 	return []byte(j.String())
 }
 
-// Add returns a new instance of JSON with the new value added.
-func (j JSON) Add(insertKey string, insertValue interface{}) (JSON, error) {
+// AsMap returns j as a map
+func (j JSON) AsMap() (map[string]interface{}, error) {
 	output := make(map[string]interface{})
-
 	switch v := j.Result.Value().(type) {
 	case map[string]interface{}:
 		for key, value := range v {
-			if key != insertKey {
-				output[key] = value
-			}
+			output[key] = value
 		}
-		output[insertKey] = insertValue
 	case nil:
-		output[insertKey] = insertValue
 	default:
-		return JSON{}, errors.New("can only add to JSON objects or null")
+		return nil, errors.New("can only add to JSON objects or null")
 	}
+	return output, nil
+}
 
-	bytes, err := json.Marshal(output)
+// mapToJSON returns m as a JSON object, or errors
+func mapToJSON(m map[string]interface{}) (JSON, error) {
+	bytes, err := json.Marshal(m)
 	if err != nil {
 		return JSON{}, err
 	}
-
 	return JSON{Result: gjson.ParseBytes(bytes)}, nil
 }
 
+// Add returns a new instance of JSON with the new value added.
+func (j JSON) Add(insertKey string, insertValue interface{}) (JSON, error) {
+	return j.MultiAdd([]KV{{insertKey, insertValue}})
+}
+
+// KV represents a key/value pair to be added to a JSON object
+type KV struct {
+	Key   string
+	Value interface{}
+}
```

Comment:
Why not use a map instead?

Reply:
You're right, that's much cleaner. I've re-arranged it that way.